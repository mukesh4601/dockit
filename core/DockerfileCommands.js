class Command {
    getCommand() {

    }
}

class RunCommand extends Command{

    constructor(command) {
        super();
        if (!(command instanceof Array)) {
            throw new Error("Expected instance of type Array but found : "+(typeof command));
        }

        this.command = command;
    }

    getCommand() {
        return "RUN " + this.command.reduce((previousValue, currentValue) => previousValue + " " + currentValue);
    }
}

class CopyCommand extends Command{
    constructor(src,dest) {
        super();
        if (typeof src !== 'string'){
            throw Error("Expected a string but found "+(typeof src))
        }
        if (typeof dest !== 'string'){
            throw Error("Expected a string but found "+(typeof dest))
        }

        this._src = src;
        this._dest = dest;
    }

    getCommand() {
        return "COPY "+this._src+ " "+this._dest
    }
}

class EntryPointCommand extends Command{

    constructor(command) {
        super();
        if (!(command instanceof Array)) {
            throw new Error("Expected instance of type Array but found : "+(typeof command));
        }

        this._command = command;
    }


    getCommand(){
        return "ENTRYPOINT " + this._command.reduce((previousValue, currentValue) => previousValue + " " + currentValue);
    }
}

export {CopyCommand,RunCommand,EntryPointCommand}