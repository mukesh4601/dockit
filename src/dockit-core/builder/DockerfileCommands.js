class Command {
    getCommand() {}
}

class RunCommand extends Command{

    constructor(command) {
        super();
        if (!(command instanceof Array)) {
            throw new Error("Expected instance of type Array but found : "+(typeof command));
        }
        if (!command.length) {
            throw new Error("Empty array")
        }

        this.command = command;
    }

    getCommand() {
        return "RUN " + this.command.reduce((previousValue, currentValue) => previousValue + " " + currentValue);
    }
}

class CopyCommand extends Command{
    constructor(src,dest) {
        super();
        if (typeof src !== 'string'){
            throw Error("Expected a string but found "+(typeof src))
        }
        if (typeof dest !== 'string'){
            throw Error("Expected a string but found "+(typeof dest))
        }

        this._src = src;
        this._dest = dest;
    }

    getCommand() {
        return "COPY "+this._src+ " "+this._dest
    }
}

class EntryPointCommand extends Command{

    constructor(command) {
        super();
        if (!(command instanceof Array)) {
            throw new Error("Expected instance of type Array but found : "+(typeof command));
        }

        this._command = command;
    }


    getCommand(){
        return "ENTRYPOINT " + this._command.reduce((previousValue, currentValue) => previousValue + " " + currentValue);
    }
}

class EnvCommand extends Command{
    constructor(command) {
        super();
        this.NAME = 0;
        this.VALUE = 1;
        if (!(command instanceof Array)) {
            throw new Error("Expected instance of type Array but found : "+(typeof command));
        }

        this.command = command;
    }

    getCommand() {
        if (this.command.length < 1) {
            return "";
        }
        return "ENV " + this.command.map((envs) => {
                return envs[this.NAME] + "="+ envs[this.VALUE]
            }).reduce((previousValue, currentValue) => previousValue + " " + currentValue);
    }
}

class CMDCommand extends Command{

}

export {CopyCommand,RunCommand,EntryPointCommand,EnvCommand}