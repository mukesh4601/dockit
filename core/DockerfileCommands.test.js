import {RunCommand,CopyCommand,EntryPointCommand,EnvCommand} from './DockerfileCommands'

test('Test Commands validation : throw exception',()=>{
    expect(()=>{new RunCommand()}).toThrow();
    expect(()=>{new CopyCommand()}).toThrow();
    expect(()=>{new CopyCommand("")}).toThrow();
    expect(()=>{new EntryPointCommand()}).toThrow();
    expect(()=>{new EnvCommand()}).toThrow();
    expect(new EnvCommand([]).getCommand()).toBe("");
});

test('Test Commands validation : valid arguments',()=>{
    expect(new RunCommand([])).toBeInstanceOf(RunCommand);
    expect(new CopyCommand("","")).toBeInstanceOf(CopyCommand);
    expect(new EntryPointCommand([])).toBeInstanceOf(EntryPointCommand);
    expect(new EnvCommand([])).toBeInstanceOf(EnvCommand);
});