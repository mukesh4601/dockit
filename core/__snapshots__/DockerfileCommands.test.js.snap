// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CopyCommand 1`] = `"COPY src dest"`;

exports[`Test EntryPointCommand 1`] = `"ENTRYPOINT java -jar java.jar"`;

exports[`Test EnvCommand 1`] = `"ENV env1=value2 env2=value2"`;

exports[`Test RunCommand 1`] = `"RUN /bin/bash echo \${SAIF}"`;
